core/src/sign.c

```diff
@@ -480,30 +481,39 @@ compute_output_hash(char xpub[static MULTISIG_PARTS][XPUB_SIZE],
  *
  * The code creates two signatures for a 2-of-4 segwit wallet. The result can't
  * be verified with Electrum but we can broadcast it on the Testnet.
+ *
+ * Confidential items have been marked.
+ * An effort has been made to zeroize them regardless of success or failure of this function.
+ *
  */
 Result handle_sign_tx(InternalCommandRequest_SignTxRequest *request,
                       InternalCommandResponse_SignTxResponse *response) {
+  Result r = Result_SUCCESS;
+  uint8_t CONFIDENTIAL seed[SHA512_DIGEST_LENGTH] = {0};
+  HDNode CONFIDENTIAL wallet;
+  memzero((void *) &wallet, sizeof(HDNode));
```

Comment:
I believe casting the pointer to `void *` in this case (and cases later in the file) is unnecessary, and strictly speaking incorrect. In the declaration of `memset`, the parameter type `void *` means it takes a generic pointer (any pointer type) as an argument.